stan/math/prim/scal/err/check_3F2_converges.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef STAN_MATH_PRIM_SCAL_ERR_CHECK_3F2_CONVERGES_HPP
+#define STAN_MATH_PRIM_SCAL_ERR_CHECK_3F2_CONVERGES_HPP
+
+#include <stan/math/prim/scal/fun/value_of_rec.hpp>
+#include <cmath>
+#include <stdexcept>
+#include <sstream>
+#include <limits>
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Check if the hypergeometric function (3F2) called with
+     * supplied arguments will converge, assuming arguments are
+     * finite values.
+     *
+     * @tparam T_a1 Type of a1
+     * @tparam T_a2 Type of a2
+     * @tparam T_a3 Type of a3
+     * @tparam T_b1 Type of b1
+     * @tparam T_b2 Type of b2
+     * @tparam T_z Type of z
+     *
+     * @param function Name of function ultimately relying on 3F2 (for error
+     &   messages)
+     * @param a1 Variable to check
```

Comment:
They are actually undifferentiated in the function definition since what matters is a product of their shifted factorials.  I think there's not much to do about it.  If you look at the function definition (will add in the function itself) it makes sense to just treat them as two groups of parameters.  They could be passed in as vectors to make it clear and then we could just have a qFp function.  I think that's better for a down-the-road refactor.